spread.yaml

```diff
@@ -66,6 +84,33 @@ prepare: |
     # this indicates that the server got reused, nothing to setup
     [ "$REUSE_PROJECT" != 1 ] || exit 0
 
+    # Disable burst limit so resetting the state quickly doesn't create problems.
+    mkdir -p /etc/systemd/system/snapd.service.d
+    cat <<EOF > /etc/systemd/system/snapd.service.d/local.conf
+    [Unit]
+    StartLimitInterval=0
+    [Service]
+    Environment=SNAPD_DEBUG_HTTP=7 SNAP_REEXEC=0
+    EOF
+
+    # adhoc backend and ubuntu-core system, nothing to setup
+    if [ "$SPREAD_BACKEND" = external-ubuntu-core ]; then
+        # build test binaries
+        if [ ! -f $GOPATH/bin/snapbuild ]; then
+            mkdir -p $GOPATH/bin
+            snap install --devmode --edge classic
+            classic "sudo apt update && apt install -y git golang-go"
+            classic "GOPATH=$GOPATH go get ..${PROJECT_PATH}/tests/lib/snapbuild"
+            snap remove classic
+        fi
+        mkdir -p /etc/systemd/system/snapd.socket.d
+        cat <<EOF > /etc/systemd/system/snapd.socket.d/local.conf
```

Comment:
Okay, we're already doing that for snapd.service, so should be okay for the socket.

Let's please do both on the same place, instead of special casing this system and backend.